### ISSUE TITLE
ST7735S: Display no longer works if RST pin is not selected

### ISSUE THREAD

**Issue by Rajssss** [lvgl/lvgl_esp32_drivers#76]
Hello, thank you for maintaining this amazing library, saved a lot of my work.
I am trying to save some pins and decided not to use RST pin, hoping that the software reset will do the job, also I have connected RST pin to 3.3V.
But it just doesn't work. I have noticed that the source (st7735s.c) has the required code missing:

```
	//Initialize non-SPI GPIOs
        gpio_pad_select_gpio(ST7735S_DC);
	gpio_set_direction(ST7735S_DC, GPIO_MODE_OUTPUT);

#if ST7735S_USE_RST
        gpio_pad_select_gpio(ST7735S_RST);
	gpio_set_direction(ST7735S_RST, GPIO_MODE_OUTPUT);

	//Reset the display
	gpio_set_level(ST7735S_RST, 0);
	vTaskDelay(100 / portTICK_RATE_MS);
	gpio_set_level(ST7735S_RST, 1);
	vTaskDelay(100 / portTICK_RATE_MS);
#endif
```

I have tried adding `st7735s_send_cmd(TFT_SWRST);`
but still don't know why it doesn't work even after that, here is how it looks:
```
	//Initialize non-SPI GPIOs
        gpio_pad_select_gpio(ST7735S_DC);
	gpio_set_direction(ST7735S_DC, GPIO_MODE_OUTPUT);

#if ST7735S_USE_RST
        gpio_pad_select_gpio(ST7735S_RST);
	gpio_set_direction(ST7735S_RST, GPIO_MODE_OUTPUT);

	//Reset the display
	gpio_set_level(ST7735S_RST, 0);
	vTaskDelay(100 / portTICK_RATE_MS);
	gpio_set_level(ST7735S_RST, 1);
	vTaskDelay(100 / portTICK_RATE_MS);
#else
	st7735s_send_cmd(TFT_SWRST);
#endif
```

Not sure what else I should do to fix this issue.

**Comment by tore-espressif** [comment]
![image](https://user-images.githubusercontent.com/77206109/123091786-0acf3a00-d42a-11eb-9383-e24f6ce04fc6.png)

Although, it should work with SW reset too.
Try adding 200ms delay before and after sending the SWreset command. I'll have a look if it doesn't help.

**Comment by Rajssss** [comment]
> ![image](https://user-images.githubusercontent.com/77206109/123091786-0acf3a00-d42a-11eb-9383-e24f6ce04fc6.png)
> 
> Although, it should work with SW reset too.
> Try adding 200ms delay before and after sending the SWreset command. I'll have a look if it doesn't help.

Didn't work. Same issue.

```
#ifdef CONFIG_LV_DISP_USE_RST
    gpio_pad_select_gpio(ST7735S_RST);
	gpio_set_direction(ST7735S_RST, GPIO_MODE_OUTPUT);

	//Reset the display
	gpio_set_level(ST7735S_RST, 0);
	vTaskDelay(100 / portTICK_RATE_MS);
	gpio_set_level(ST7735S_RST, 1);
	vTaskDelay(100 / portTICK_RATE_MS);
#else
	st7735s_send_cmd(ST7735_SWRESET);
	vTaskDelay(pdMS_TO_TICKS(200));
#endif
```

**Comment by C47D** [comment]
I think I got one of this displays, will try to replicate the issue and make it work with software reset.

**Comment by Rajssss** [comment]
> ![image](https://user-images.githubusercontent.com/77206109/123091786-0acf3a00-d42a-11eb-9383-e24f6ce04fc6.png)
> 
> Although, it should work with SW reset too.
> Try adding 200ms delay before and after sending the SWreset command. I'll have a look if it doesn't help.

Didn't help.

```
#ifdef CONFIG_LV_DISP_USE_RST
    gpio_pad_select_gpio(ST7735S_RST);
	gpio_set_direction(ST7735S_RST, GPIO_MODE_OUTPUT);

	//Reset the display
	gpio_set_level(ST7735S_RST, 0);
	vTaskDelay(100 / portTICK_RATE_MS);
	gpio_set_level(ST7735S_RST, 1);
	vTaskDelay(100 / portTICK_RATE_MS);
#else
	st7735s_send_cmd(ST7735_SWRESET);
	vTaskDelay(pdMS_TO_TICKS(200));
#endif
	
```

**Comment by Rajssss** [comment]
> ![image](https://user-images.githubusercontent.com/77206109/123091786-0acf3a00-d42a-11eb-9383-e24f6ce04fc6.png)
> 
> Although, it should work with SW reset too.
> Try adding 200ms delay before and after sending the SWreset command. I'll have a look if it doesn't help.

Didn't help.

```
#ifdef CONFIG_LV_DISP_USE_RST
    gpio_pad_select_gpio(ST7735S_RST);
	gpio_set_direction(ST7735S_RST, GPIO_MODE_OUTPUT);

	//Reset the display
	gpio_set_level(ST7735S_RST, 0);
	vTaskDelay(100 / portTICK_RATE_MS);
	gpio_set_level(ST7735S_RST, 1);
	vTaskDelay(100 / portTICK_RATE_MS);
#else
	vTaskDelay(pdMS_TO_TICKS(200));
	st7735s_send_cmd(ST7735_SWRESET);
	vTaskDelay(pdMS_TO_TICKS(200));
#endif
```

**Comment by Rajssss** [comment]
@C47D any update on this?

**Comment by C47D** [comment]
No, sorry, a storm hit my city and just today we're getting back to normal. Will let you know any update over the weekend.

**Comment by Rajssss** [comment]
@C47D @tore-espressif It seems to be working, but I had to connect the RST pin to the EN pin on my ESP32, and reduce the delay to 50ms. It didn't work if the RST pin is floating/VCC/GND. Also if I reset the esp32, sometimes the display works sometimes not, I don't know why.

**Comment by C47D** [comment]
According to this image, the RST signal should go to Vcc (using a pull-up resistor)

![imagen](https://user-images.githubusercontent.com/11037705/123518683-a05a0c00-d66c-11eb-82c6-420ec5da925d.png)

**Comment by Rajssss** [comment]
> According to this image, the RST signal should go to Vcc (using a pull-up resistor)
> 
> ![imagen](https://user-images.githubusercontent.com/11037705/123518683-a05a0c00-d66c-11eb-82c6-420ec5da925d.png)

Don't know why but my display refused to work this way, I have to connect it to EN pin directly, but it works fine.
So It should be added in the code.

**Comment by C47D** [comment]
Which EN pin do you mean?

**Comment by Rajssss** [comment]
> Which EN pin do you mean?

The Enable pin on ESP32.
